openmc/mgxs_library.py

```diff
@@ -379,200 +447,437 @@ def tabular_legendre(self, tabular_legendre):
+        # checking
+        npfission = np.asarray(fission)
+        check_value('fission shape', npfission.shape, [self.vector_shape])
 
-    @k_fission.setter
-    def k_fission(self, k_fission):
-        if self._representation is 'isotropic':
-            shape = (self._energy_groups.num_groups,)
-        elif self._representation is 'angle':
-            shape = (self._num_polar, self._num_azimuthal,
-                     self._energy_groups.num_groups)
-        # check we have a numpy list
-        check_type("k_fission", k_fission, np.ndarray, expected_iter_type=Real)
-        if k_fission.shape == shape:
-            self._k_fission = np.copy(k_fission)
-            if np.sum(self._k_fission) > 0.0:
-                self._fissionable = True
-        else:
-            msg = 'Shape of provided k_fission "{0}" does not match shape ' \
-                  'required, "{1}"'.format(k_fission.shape, shape)
-            raise ValueError(msg)
+        self._fission = npfission
+
+        if np.sum(self._fission) > 0.0:
+            self._fissionable = True
+
+    @kappa_fission.setter
+    def kappa_fission(self, kappa_fission):
+        check_type('kappa_fission', kappa_fission, Iterable,
+                   expected_iter_type=Real)
+        # Convert to a numpy array so we can easily get the shape for
+        # checking
+        npkappa_fission = np.asarray(kappa_fission)
+        check_value('kappa fission shape', npkappa_fission.shape,
+                    [self.vector_shape])
+
+        self._kappa_fission = npkappa_fission
+
+        if np.sum(self._kappa_fission) > 0.0:
+            self._fissionable = True
 
     @chi.setter
     def chi(self, chi):
-        if not self._use_chi:
-            msg = 'Providing chi when nu_fission already provided as matrix!'
-            raise ValueError(msg)
-        if self._representation is 'isotropic':
-            shape = (self._energy_groups.num_groups,)
-        elif self._representation is 'angle':
-            shape = (self._num_polar, self._num_azimuthal,
-                     self._energy_groups.num_groups)
-        # check we have a numpy list
-        check_type("chi", chi, np.ndarray, expected_iter_type=Real)
-        if chi.shape == shape:
-            self._chi = np.copy(chi)
-        else:
-            msg = 'Shape of provided chi "{0}" does not match shape ' \
-                  'required, "{1}"'.format(chi.shape, shape)
-            raise ValueError(msg)
+        if self._use_chi is not None:
+            if not self._use_chi:
+                msg = 'Providing chi when nu_fission already provided as a' \
+                      'matrix'
+                raise ValueError(msg)
+
+        check_type('chi', chi, Iterable, expected_iter_type=Real)
+        # Convert to a numpy array so we can easily get the shape for
+        # checking
+        npchi = np.asarray(chi)
+        check_value('chi shape', npchi.shape, [self.vector_shape])
```

Comment:
This is an abuse of `check_value` in my opinion. I think it's better to explicitly check the shape here and then give a more meaningful error message if the shape is not as expected.